baseplate/lib/secrets.py

```diff
@@ -509,8 +595,9 @@ def secrets_store_from_config(
         backoff = None
 
     if options.provider == "vault_csi":
-        parser = parse_vault_csi
-        return DirectorySecretsStore(options.path, parser, timeout=timeout, backoff=backoff)
+        return VaultCSISecretsStore(
+            options.path, parser=parse_vault_csi, timeout=timeout, backoff=backoff
+        )
```

Comment:
@chriskuehl We are a little torn here WRT backward incompatibility. WDYT about removing the old implementation straight away?

Reply:
PS If we decide to replace the current implementation, we can probably remove DirectorySecretsStore & associated tests.

Reply:
This would be transparent to users, right? Looking in Sourcegraph, I don't see any references to `DirectorySecretsStore` outside of this repo, so that seems reasonable to me.

Reply:
Internally we seem fine.

The issue is that theoretically there are open source users of baseplate.py that might see breakage. Is it ok to risk that?

Reply:
I think it's OK to remove the old unused implementation. I did some cursory checks (searching GitHub for public users of baseplate) and also checking in with the rest of the team, and our general impression is that nobody is really using Baseplate.py outside of Reddit.

This isn't necessarily great open source hygiene, but I don't think it's worth increasing our maintenance burden for a benefit we're pretty sure isn't actually there. We should call out in the release notes that it's technically a breaking change though.

Reply:
Ok, I've ripped out the old implementation based on this discussion.